src/components/events/EventCard.jsx

```diff
@@ -1,17 +1,47 @@
-import { Link } from "react-router-dom";
+import { useSelector } from "react-redux";
+import { selectToken } from "../auth/authSlice";
+import { useUpdateEventMutation } from "./eventsSlice";
+import { useState, useEffect } from "react";
+import { jwtDecode } from "jwt-decode";
 
-//Tooken import and selector
-// import { useSelector } from "react-redux";
-// import { selectToken } from "../../store/authSlice";
+// import { Link } from "react-router-dom";
 
 export default function EventCard({ event }) {
   const date = event.startTime.slice(0, 10);
   const time = event.startTime.slice(11, 19);
 
-  // const token = useSelector(selectToken);
+  const token = useSelector(selectToken);
 
-  const token = true;
-  const attending = true;
+  //state for changing button classnames when going or not going
+  const [attending, setAttending] = useState(false);
+
+  /*Using use effect to only run the logic when the user changes or the attending users
+   Change to run the code only when its needed*/
+  useEffect(() => {
+    if (token) {
+      // Decode token to get user ID
+      //Extracts the id of the the decoded users token
+      const { id: userId } = jwtDecode(token);
+      const isAttending = event.attendingUsers?.some(
+        (user) => user.id === userId
+      );
+      setAttending(isAttending);
+    }
+  }, [token, event.attendingUsers]);
+
+  const [updateEvent] = useUpdateEventMutation();
+
+  const toggleAttendance = async (newStatus) => {
+    try {
+      await updateEvent({
+        id: event.id,
+        attending: newStatus,
+      });
+      setAttending(newStatus);
+    } catch (error) {
+      console.error("Error updating attendance:", error);
+    }
+  };
```

Comment:
The req.user.id would only work in this async function and I needed it on my state, so I used useEffect and decode. :)